tensorboard/webapp/webapp_data_source/tb_feature_flag_query_parameters.ts

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {FeatureFlags} from '../feature_flag/types';
+import {
+  ENABLE_CARD_WIDTH_SETTING_PARAM_KEY,
+  ENABLE_COLOR_GROUP_BY_REGEX_QUERY_PARAM_KEY,
+  ENABLE_COLOR_GROUP_QUERY_PARAM_KEY,
+  ENABLE_DARK_MODE_QUERY_PARAM_KEY,
+  ENABLE_DATA_TABLE_PARAM_KEY,
+  ENABLE_LINKED_TIME_PARAM_KEY,
+  EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY,
+  FORCE_SVG_RENDERER,
+  SCALARS_BATCH_SIZE_PARAM_KEY,
+} from './tb_feature_flag_data_source_types';
+
+export type BaseFeatureFlagType = Boolean | Number | String | null | undefined;
```

Comment:
Instead of BaseFeatureFlagType and FeatureFlagType, my instinct is to instead make FeatureFlagMetadata a generic type:

export type FeatureFlagMetadata<T> = {
  displayName: string;
  queryParamOverride?: string;
  parseValue: (str: string) => T;
};

And then sort of adjust things from there. I've done this here: 

https://github.com/bmd3k/tensorboard/commit/c1060ca992afee047628c118a22d85d6ff11a9d2

It compiles and runs but it is incomplete -- it doesn't handle the experimentalPlugins case and and there are some parts of the code that we have to cast things to, sort of, bypass the compiler. I think it is more readable, though and still largely gives good compiler coverage.

Reply:
I originally had done this, then someplace in the middle of a refactor couldn't get it to work. Thanks for bringing this back up.